src/Algebra/Definitions/RawMagma.agda

```diff
@@ -54,15 +56,38 @@ _∤_ : Rel A (a ⊔ ℓ)
 x ∤ y = ¬ x ∣ y
 
 ------------------------------------------------------------------------
--- Mutual divisibility
-
--- When in a cancellative monoid, elements related by _∣∣_ are called
--- associated and if x ∣∣ y then x and y differ by an invertible factor.
-
-infix 5 _∣∣_ _∤∤_
+-- Mutual divisibility _∣∣_.
+-- In a  monoid,  this is an equivalence relation extending _≈_.
+-- When in a  cancellative monoid,  elements related by _∣∣_ are called
+-- associated,  and there x ∣∣ y is equivalent to that x and y differ by
+-- some invertible factor.
+-- Example: for ℕ  this is equivalent to  x ≡ y,
+--          for ℤ  this is equivalent to  (x ≡ y  or  x ≡ -y).
 
 _∣∣_ : Rel A (a ⊔ ℓ)
 x ∣∣ y = x ∣ y × y ∣ x
 
 _∤∤_ : Rel A (a ⊔ ℓ)
-x ∤∤ y =  ¬ x ∣∣ y
+x ∤∤ y = ¬ x ∣∣ y
+
+------------------------------------------------------------------------
+-- Greatest common divisor (GCD)
+
+record IsGCD (x y gcd : A) : Set (a ⊔ ℓ) where
+  constructor mkIsGCD
+  field
+    divides₁ : gcd ∣ x
+    divides₂ : gcd ∣ y
+    greatest : ∀ {z} → z ∣ x → z ∣ y → z ∣ gcd
+
+  quot₁ : A               -- a complementory quotient  x/gcd
```

Comment:
I don't think there's any need for these extra projections, as you can always extract them via pattern matching.

Reply:
These lines starting from No 82 are removed.